lib/house.rb

```diff
@@ -0,0 +1,52 @@
+class House
+  attr_reader :price, :address, :rooms
+
+  def initialize(price, address)
+    @price = price.tr('^0-9', '').to_i
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def above_market_average?
+    @price > 500_000
+  end
+
+  def rooms_from_category(category)
+    @rooms.select { |room| room.category == category }
+  end
+
+  def area
+    @rooms.map { |room| room.area }.sum
```

Comment:
remember, sum is an enumerable too. 

```
def area
    @room.sum { |room| room.area }
end
```

The way you have it, you're technically enumerating twice (one for map and one for sum), which is unnecessary.